### DISCUSSION TITLE
Mouse Follows Focus

### DISCUSSION THREAD

**Discussion by distek** [pop-os/shell#1201] [Q&A]
Is there a way to do this? This would be if the focus changes, the mouse would move to the center(or wherever) of the window that is now focused. An extension or otherwise.

To be clear, this is not focus follows mouse. That being when you mouse over a window, it gets focused. I already have this enabled.

**Comment by leviport** [comment]
You're not seeing this behavior already? The default behavior is to move the mouse cursor to the top left corner (nearest to most menu dropdowns on most windows) when focus is changed with the launcher or with Super + arrow keys. How are you changing focus where this doesn't happen?

**Reply by mmstick** [reply]
Perhaps "focus follows mouse" is overriding that behavior

**Reply by leviport** [reply]
I almost always have "Focus on Hover" enabled in Tweaks > Windows > Window Focus, and I don't think I've ever seen the cursor not move when I change focus with hotkeys or the launcher. 

@distek are you running pop-shell on Pop!\_OS? I'm wondering if using pop-shell on a different distro would somehow be a factor.

**Reply by distek** [reply]
I was talking with some folks on Matrix last night and we believe it's because I'm using Wayland.

Mostly stock gnome with the exception of pop-shell and on Arch Linux.

From what I gathered, this functionality might not be available on Gnome-Wayland just yet. I tried moving the mouse cursor with ydotool and that doesn't appear to be working either.

**Reply by mmstick** [reply]
I'm only using GNOME's internal Mutter APIs for moving the mouse, so it should theoretically be working in Wayland, but I know that some functionality in Mutter is broken in Wayland, like this.

**Reply by distek** [reply]
Would you like me to file an issue for this? Or will this just come in time once they get Wayland more sorted out upstream?

**Reply by mangkoran** [reply]
Recently installed `pop-shell` on GNOME 42.2 (wayland), Ubuntu 22.04 LTS and I also encounter this issue.

**Reply by leviport** [reply]
@mangkoran You can turn that off now if you want. Go to pop-shell's settings from Extensions and turn off "Mouse cursor follows active window".

**Comment by jbtrystram** [comment]
This seems to still be not working in Fedora 37 with Gnome 43

**Comment by heinosasshallik** [comment]
It's also not working for me.

```
lsb_release -a && gnome-shell --version
No LSB modules are available.
Distributor ID: Pop
Description:    Pop!_OS 22.04 LTS
Release:        22.04
Codename:       jammy

GNOME Shell 42.9
```

It's pretty annoying. I tried to get the functionality working with [this](https://github.com/LeonMatthes/mousefollowsfocus) extension but that one doesn't take PopOS's peculiarities into account (stacked windows, etc).
